File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        Set<Integer> s = new HashSet<>();
        for (int n : nums) s.add(n);
        return s.size() > nums.length / 2 ? nums.length / 2 : s.size();
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <set>
using namespace std;

int solution(vector<int> nums)
{
    set<int> s;
    s.insert(nums.begin(), nums.end());
    return s.size() > nums.size() / 2 ? nums.size() / 2 : s.size();
}